test/integration/smart_answer_flows/calculate_your_holiday_entitlement_test.rb

```diff
@@ -0,0 +1,387 @@
+      assert_equal :irregular_and_annualised_done, next_node
+    end
+
+    should "take annualised-hours to :irregular_and_annualised_done" do
+      responses[:basis_of_calculation?] = "annualised-hours"
+      assert_equal :irregular_and_annualised_done, next_node
+    end
+
+    should "take shift-worker to :shift_worker_hours_per_shift?" do
+      responses[:basis_of_calculation?] = "shift-worker"
+      assert_equal :shift_worker_hours_per_shift?, next_node
+    end
+  end
+
+  context "question :how_many_hours_per_week?" do
+    setup do
+      responses = {
+        basis_of_calculation?: "hours-worked-per-week",
+        calculation_period?: "leaving",
+        what_is_your_starting_date?: "2021-01-01",
+        what_is_your_leaving_date?: "2021-10-01",
+      }
+    end
+
+    should see_question(:how_many_hours_per_week?).with_title("Number of hours worked per week?")
+
+    context "invalid no of hours given" do
+      should_not accept_answer(-1)
+      should_not accept_answer(169)
+    end
+
+    should accept_answer(100).with_next_node(:how_many_days_per_week_for_hours?)
+  end
+
+  context "question :how_many_days_per_week_for_hours?" do
+    setup do
+      responses = {
+        basis_of_calculation?: "hours-worked-per-week",
+        calculation_period?: "leaving",
+        what_is_your_starting_date?: "2021-01-01",
+        what_is_your_leaving_date?: "2021-10-01",
+        how_many_hours_per_week?: "40",
+      }
+    end
+
+    should see_question(:how_many_days_per_week_for_hours?).with_title("Number of days worked per week?")
+
+    context "invalid no of hours given" do
+      should_not accept_answer(8)
+      should_not accept_answer(1)
+    end
+
+    should accept_answer(5).with_next_node(:hours_per_week_done)
+
+    should "take compressed-hours to :compressed_hours_done" do
+      responses[:basis_of_calculation?] = "compressed-hours"
+      assert_equal :compressed_hours_done, next_node
+    end
+  end
+
+  context "question :shift_worker_basis?" do
+    setup do
+      responses = { basis_of_calculation?: "shift-worker" }
+    end
+
+    should see_question(:shift_worker_basis?).with_title("Do you want to calculate the holiday:")
+
+    should accept_answer("full-year").with_next_node(:shift_worker_hours_per_shift?)
+    should accept_answer("starting").with_next_node(:what_is_your_starting_date?)
+    should accept_answer("starting-and-leaving").with_next_node(:what_is_your_starting_date?)
+    should accept_answer("leaving").with_next_node(:what_is_your_leaving_date?)
+  end
+
+  context "question :shift_worker_hours_per_shift?" do
+    setup do
+      responses = {
+        basis_of_calculation?: "shift-worker",
+        shift_worker_basis?: "full-year",
+      }
+    end
+
+    should see_question(:shift_worker_hours_per_shift?).with_title("How many hours in each shift?")
+
+    context "invalid no of hours given" do
+      should_not accept_answer(-1)
+      should_not accept_answer(25)
+    end
```

Comment:
There may be more of these too `should_not_accept_invalid_hours_per_week` for example.  Will have a play with that idea.

Reply:
I've added the following proposed methods - they do look so much better 👍 

```ruby
should_not_accept_a_negative_response
should_not_accept_an_invalid_no_of_days_per_week
should_not_accept_an_invalid_no_of_hours_per_week
should_not_accept_an_invalid_no_of_hours_per_day
should_not_accept_a_response_before(:a_date_or_value)
should_not_accept_a_response_more_than_a_year_from(:a_date)
```
_Note_: I'm assuming that `should_not_accept_a...` is based on the current response, so it doesn't need to be explicitly given.

Reply:
Thanks Graham, they do read better. I think you've gone a bit far with them though to use them for every validation scenario.

I'd have thought we'd only want to create a helper method to deal with repetitive scenarios - otherwise the helper methods act a bit like misdirection. Could we cut them back to only be used in cases where the same validation is done 3 times or more?

Reply:
Fixed - only the last two in the list above were kept as they are the only ones used 3 or more times.